Formula/g/gnu-cobol.rb

```diff
@@ -20,40 +21,87 @@ class GnuCobol < Formula
     sha256 x86_64_linux:   "4bf957f586d155f1034ac90ac24edff1253e9de935336680689d85f0bbdcf45c"
   end
 
+  head do
+    url "https://svn.code.sf.net/p/gnucobol/code/trunk"
+
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "bison" => :build
+    depends_on "flex" => :build
+    depends_on "gettext" => :build
+    depends_on "help2man" => :build
+    depends_on "libtool" => :build
+    depends_on "texinfo" => :build
+
+    # GnuCOBOL 4.x+ (= currently only on head) features more backends
+    # and multiple indexed handlers, so we add dependencies for those here
+    depends_on "lmdb"
+    depends_on "unixodbc"
+    # TODO: add "visam" and --with-visam, once formula is added
+  end
+
+  depends_on "pkg-config" => :build
   depends_on "berkeley-db"
   depends_on "gmp"
+  depends_on "json-c"
 
-  def install
-    # both environment variables are needed to be set
-    # the cobol compiler takes these variables for calling cc during its run
-    # if the paths to gmp and bdb are not provided, the run of cobc fails
-    gmp = Formula["gmp"]
-    bdb = Formula["berkeley-db"]
-    ENV.append "CPPFLAGS", "-I#{gmp.opt_include} -I#{bdb.opt_include}"
-    ENV.append "LDFLAGS", "-L#{gmp.opt_lib} -L#{bdb.opt_lib}"
+  uses_from_macos "libxml2"
+  uses_from_macos "ncurses"
 
+  def install
     # Avoid shim references in binaries on Linux.
     ENV["LD"] = "ld" unless OS.mac?
 
-    system "./configure", "--disable-debug",
-                          "--disable-dependency-tracking",
-                          "--disable-silent-rules",
-                          "--prefix=#{prefix}",
-                          "--with-libiconv-prefix=/usr",
-                          "--with-libintl-prefix=/usr"
+    args = %w[
+      --disable-silent-rules
+      --with-libiconv-prefix=/usr
+      --with-libintl-prefix=/usr
+      --with-json=json-c
+      --with-xml2
+      --with-curses=ncurses
+      --with-db
+    ]
+
+    if build.head?
+      # GnuCOBOL 4.x+ features (= currently only on head)
+      args += %w[
+        --with-lmdb
+        --with-odbc
+        --with-indexed=db
+      ]
+    end
+
+    # for head: bootstrap, go with whatever gettext we have
+    if build.head?
```

Comment:
Let's avoid two consecutive `if`'s having the same condition.

```suggestion

      # bootstrap, go with whatever gettext we have
```